lib/resourcemerge/core.go

```diff
@@ -774,12 +764,10 @@ func setInt64(modified *bool, existing *int64, required int64) {
 }
 
 func setInt64Ptr(modified *bool, existing **int64, required *int64) {
-	// if we have no required, then we don't care what someone else has set
-	if required == nil {
+	if *existing == nil && required == nil {
```

Comment:
This change will also tighten some other places:

```console
$ git grep '^func \|setInt64Ptr' origin/pr/804 | grep -B1 setInt64Ptr
origin/pr/804:lib/resourcemerge/batch.go:func EnsureJob(modified *bool, existing *batchv1.Job, required batchv1.Job) {
origin/pr/804:lib/resourcemerge/batch.go:       setInt64Ptr(modified, &existing.Spec.ActiveDeadlineSeconds, required.Spec.ActiveDeadlineSeconds)
--
origin/pr/804:lib/resourcemerge/core.go:func ensureSecurityContext(modified *bool, existing *corev1.SecurityContext, required corev1.SecurityContext) {
origin/pr/804:lib/resourcemerge/core.go:        setInt64Ptr(modified, &existing.RunAsUser, required.RunAsUser)
--
origin/pr/804:lib/resourcemerge/core.go:func ensurePodSecurityContext(modified *bool, existing *corev1.PodSecurityContext, required corev1.PodSecurityContext) {
origin/pr/804:lib/resourcemerge/core.go:        setInt64Ptr(modified, &existing.RunAsUser, required.RunAsUser)
origin/pr/804:lib/resourcemerge/core.go:        setInt64Ptr(modified, &existing.RunAsGroup, required.RunAsGroup)
origin/pr/804:lib/resourcemerge/core.go:        setInt64Ptr(modified, &existing.FSGroup, required.FSGroup)
--
origin/pr/804:lib/resourcemerge/core.go:func ensureTerminationGracePeriod(modified *bool, existing **int64, required *int64) {
origin/pr/804:lib/resourcemerge/core.go:        setInt64Ptr(modified, existing, required)
--
origin/pr/804:lib/resourcemerge/core.go:func setInt64(modified *bool, existing *int64, required int64) {
origin/pr/804:lib/resourcemerge/core.go:func setInt64Ptr(modified *bool, existing **int64, required *int64) {
```

That's probably fine, but it may be worth calling out Job's `spec.activeDeadlineSeconds` and similar in the commit message.

Reply:
added

Reply:
```console
$ git grep 'func \|ensureTerminationGracePeriod' lib | grep -B1 ensureTerminationGracePeriod
lib/resourcemerge/core.go:func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.PodSpec) {
lib/resourcemerge/core.go:      ensureTerminationGracePeriod(modified, &existing.TerminationGracePeriodSeconds, required.TerminationGracePeriodSeconds)
--
lib/resourcemerge/core.go:func ensureDNSPolicy(modified *bool, existing *corev1.DNSPolicy, required corev1.DNSPolicy) {
lib/resourcemerge/core.go:func ensureTerminationGracePeriod(modified *bool, existing **int64, required *int64) {
```

This will touch all pods, not just the "Other aspects of Job" that dcbfb8272baca is calling out, but you do have an "e.g." in there, so :shrug: I don't mind if you explicitly point out that all these pod-wrapping types will now always have opinions on `terminationGracePeriod`, etc.